### DISCUSSION TITLE
Emitting to server in on_connect raises exception

### DISCUSSION THREAD

**Discussion by billysan** [miguelgrinberg/python-socketio#1371] [Q&A]
This is the gist of it:

- I have known to have a shakey and laggy connection between client and server
- I try to `client.emit()` immediately upon connection in `client.on_connect`
- Occasionally I get `BadNamespaceError` for having the namespace not yet connected
- Putting a `sleep(1)` before emitting solves the issue
- I didn't pass the `wait` parameter so it should be `True`
- Namespace is default `/`

Questions:
- Why is the namespace not yet connected after `client.on_connect`?
- is calling `sleep` a reasonable solution?

Sorry for not supplying code context I'm on the road and AFK (will be for a while)

**Comment by miguelgrinberg** [comment]
I need to see your code when you can provide it, please.
